src/batch/src/executor/join/hash_join.rs

```diff
@@ -2018,13 +2017,17 @@ impl DataChunkMutator {
                 new_visibility.set(row_id, true);
             }
         }
-        if !has_more_output_rows && !*found_non_null {
-            new_visibility.set(start_row_id, true);
+        if !has_more_output_rows {
+            if !*found_non_null {
+                new_visibility.set(start_row_id, true);
+            }
+            *found_non_null = false;
         }
 
         first_output_row_ids.clear();
 
-        self.0.set_visibility(new_visibility.finish());
+        self.0
+            .set_visibility(new_visibility.finish() & self.0.visibility());
```

Comment:
I think it is not nessarry because the input chunk's visibility shoule be empty. Though the code is not clear but I think it will not produce correctness issue

Reply:
Yes for this use case, it won't affect the correctness, tut for this method, it is necessary.